From d8869c110dfa90f8b4b287fe7543c305ad201b1d Mon Sep 17 00:00:00 2001
From: Veaceslav Falico <vfalico@redhat.com>
Date: Thu, 21 Aug 2014 11:28:33 -0400
Subject: [ethernet] netxen: slight optimization of addr compare

Message-id: <1408620515-703-9-git-send-email-vfalico@redhat.com>
Patchwork-id: 88518
O-Subject: [PATCH RHEL7 08/10] net: netxen: slight optimization of addr compare
Bugzilla: 1089356
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>

RHBZ#1089356

Testing:
Locally and by QLogic (which supplied the patch)

Description:
Backport of the following commit:

commit 202af853ccdf1c23e81d1899145d21e9a4c6d5a6
Author: dingtianhong <dingtianhong@huawei.com>
Date:   Mon Dec 30 15:41:10 2013 +0800

    net: netxen: slight optimization of addr compare

    Use possibly more efficient ether_addr_equal
    to instead of memcmp.

    Cc: Manish Chopra <manish.chopra@qlogic.com>
    Cc: Sony Chacko <sony.chacko@qlogic.com>
    Cc: Rajesh Borundia <rajesh.borundia@qlogic.com>
    Signed-off-by: Ding Tianhong <dingtianhong@huawei.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/qlogic/netxen/netxen_nic_hw.c b/drivers/net/ethernet/qlogic/netxen/netxen_nic_hw.c
index 67efe75..79f7c39 100644
--- a/drivers/net/ethernet/qlogic/netxen/netxen_nic_hw.c
+++ b/drivers/net/ethernet/qlogic/netxen/netxen_nic_hw.c
@@ -663,7 +663,7 @@ static int nx_p3_nic_add_mac(struct netxen_adapter *adapter,
  list_for_each(head, del_list) {
   cur = list_entry(head, nx_mac_list_t, list);
 
-  if (memcmp(addr, cur->mac_addr, ETH_ALEN) == 0) {
+  if (ether_addr_equal(addr, cur->mac_addr)) {
    list_move_tail(head, &adapter->mac_list);
    return 0;
   }
-- 
1.7.1